### ISSUE TITLE
Distributed validation with Keras and Tensorflow

### ISSUE THREAD

**Issue by pbruneau** [horovod/horovod#3535]
**Is your feature request related to a problem? Please describe.**
Referring to the Keras+Tensorflow example at https://github.com/horovod/horovod/blob/master/examples/tensorflow2/tensorflow2_keras_mnist.py, I'm considering distinct training and validation datasets by expanding l42 as follows:
```
train_dataset = train_dataset.repeat().shuffle(10000).batch(128)
val_dataset = val_dataset.batch(128)
```
And the adapted ``model.fit`` call as follows:
```
mnist_model.fit(train_dataset, steps_per_epoch=500 // hvd.size(), callbacks=callbacks, 
    validation_data=val_dataset, validation_steps=val_steps, epochs=24, verbose=verbose)
```

When running this code on a cluster, the model training phase uses all computation nodes as expected, but the validation phase operates only on the rank 0 node. When the validation set is pretty large, validation may take even more time than actual training.

**Describe the solution you'd like**
Ideally, validation would use all computation nodes under the hood.

**Describe alternatives you've considered**
I could not find any working solution which did not require abandonning Keras. As the optimizer object is the cornerstone for data distribution with Keras+Tensorflow, I have little idea on cues to investigate. [This page mentions a distributed validation step](http://www.idris.fr/eng/jean-zay/gpu/jean-zay-gpu-hvd-tf-multi-eng.html#distributed_validation), but I could not figure out a way to incorporate it in a working custom validation step made following [the official Tensorflow guidelines](https://www.tensorflow.org/guide/keras/customizing_what_happens_in_fit#providing_your_own_evaluation_step).
